packages/nameguard-js/src/data/canonicals.ts

```diff
@@ -0,0 +1,27 @@
+// for compression the json type is:
+// { [key: string]: [string, number] }
+import GRAPHEME_CANONICALS_ from "./canonicals.json";
+
+/**
+ * Stores information about a potentially confusable grapheme and its canonical form.
+ */
+export interface GraphemeCanonical {
+  /**
+   * The grapheme that is considered the canonical form of this grapheme.
+   */
+  canonicalGrapheme: string;
+
+  /**
+   * The number of other graphemes that this grapheme is confusable with.
+   * Always greater than or equal to 0.
+   */
+  numConfusables: number;
+};
+
+/**
+ * Map containing graphemes and their canonical forms.
+ */
+export const GRAPHEME_CANONICALS: Map<string, GraphemeCanonical> =
+  new Map(
+    Object.entries(GRAPHEME_CANONICALS_ as { [key: string]: [string, number] })
```

Comment:
Not sure, but would the following be appropriate?

```suggestion
    Object.entries(GRAPHEME_CANONICALS_ as { [key: string]: GraphemeCanonical })
```

Either way, suggest to replace all of `{ [key: string]: [string, number] }` here with a named interface definition and then making use of those named interface fields in the code below.

Reply:
I don't think so, but I'm no typescript expert. GRAPHEME_CANONICALS_ is the raw json where values are tuples, not objects. This line is just for json parsing.